### DISCUSSION TITLE
How do I generate custom response when proxy error

### DISCUSSION THREAD

**Discussion by ColinFay** [chimurai/http-proxy-middleware#862] [Q&A] [answered]
The following code will fail : 

```node
const { createProxyMiddleware } = require('http-proxy-middleware');

require('express')().use(
  '/',
  createProxyMiddleware({
    target: 'http://failhtis/'
  })
).listen(3000);
```

This will display this in my browser : 

![Screenshot 2022-12-20 at 17 10 49](https://user-images.githubusercontent.com/17936236/208713046-c54298b2-d225-44ff-ae1d-d3aa5b607ba5.png)

And in my console : 

```
[HPM] Error occurred while proxying request localhost:3000/ to http://failhtis/ [ENOTFOUND] (https://nodejs.org/api/errors.html#errors_common_system_errors)
```

What I would love to do is generate a custom message / html for the error. How do I manage that? 

I've tried the `ejectPlugin` method described here : https://github.com/chimurai/http-proxy-middleware#ejectplugins-boolean-default-false but without success. 

To sum up: I know error will happen in my setup, how do I send a custom answer instead of the "Error occured while proxying..." message?

**Comment by ColinFay** [comment, marked as answer]
I just found the answer to my question, I wasn't looking into the doc of the correct version. 

Here it is for v2

```
const { createProxyMiddleware } = require('http-proxy-middleware');

console.log(Options)

require('express')().use(
  '/',
  createProxyMiddleware({
    target: 'http://sc_rstudio/',
    changeOrigin: true,
    logLevel: 'debug',
    onError: (err, req, res) => {
      console.log("pouet")
      res.send('<h1>Something went wrong.</h1>');
    },
  })
).listen(3000);

```
